### PR TITLE
[Homework #18] - React UI by Stanisław Królikiewicz

### PR DESCRIPTION
**Pull request by stkrolikiewicz**

### CODE REVIEW

File: homeworks/stanislaw-krolikiewicz/18-react-ui/src/pages/index.tsx [hunk 1]
```diff
@@ -0,0 +1,88 @@
+import {Basic} from 'next/font/google';
+import {
+  Button,
+  NewTileButton,
+  Tile,
+  Loader,
+  Error as MyError,
+} from '@/components';
+import {ReactElement, use, useEffect, useState} from 'react';
```
Comment (VitalikMelnyk) [inline homeworks/stanislaw-krolikiewicz/18-react-ui/src/pages/index.tsx:9]: Remove `use` as a redundant import.

File: homeworks/stanislaw-krolikiewicz/18-react-ui/src/pages/index.tsx [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+});
+
+export default function Home() {
+  const [error, setError] = useState<string | null>(null);
+  const [loading, setLoading] = useState<boolean>(false);
+  const [tiles, setTiles] = useState<ReactElement[]>([]);
+
+  const addNewTile = () => {
+    setLoading(true);
+    fetch('https://tinyfac.es/api/data?limit=1&quality=0')
```
Comment (VitalikMelnyk) [inline homeworks/stanislaw-krolikiewicz/18-react-ui/src/pages/index.tsx:24]: Create a constants file where you can save all constants.
In this case, you can save the URL as a constant and use it everywhere you need.

File: homeworks/stanislaw-krolikiewicz/18-react-ui/src/pages/index.tsx [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+} from '@/components';
+import {ReactElement, use, useEffect, useState} from 'react';
+import {Avatar} from '@/types';
+import {AnimatePresence} from 'framer-motion';
+const basic = Basic({
+  weight: '400',
+  subsets: ['latin'],
+});
+
+export default function Home() {
```
Comment (VitalikMelnyk) [inline homeworks/stanislaw-krolikiewicz/18-react-ui/src/pages/index.tsx:17]: Please, rewrite all functions with async/await approach.

File: homeworks/stanislaw-krolikiewicz/18-react-ui/src/pages/index.tsx [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+  }, [error]);
+  return (
+    <main
+      className={`${basic.className} p-[40px] w-full flex flex-col items-center`}
+    >
+      <div className="flex gap-[40px] w-full flex-wrap justify-center">
+        {tiles.map((tile: ReactElement) => tile)}
+        <NewTileButton onClick={addNewTile} />
+      </div>
+      <Button onClick={fetchAvatars} className="fixed bottom-[24px] z-50">
```
Comment (VitalikMelnyk) [inline homeworks/stanislaw-krolikiewicz/18-react-ui/src/pages/index.tsx:79]: Move to a separate component.

File: homeworks/stanislaw-krolikiewicz/18-react-ui/src/components/Tile/Component.tsx [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+  avatarUrl?: string;
+}
+
+export default ({avatarUrl: iAvatar}: Props) => {
+  const [avatar, setAvatar] = useState<string | null>(iAvatar || null);
+  const [loading, setLoading] = useState<boolean>(true);
+  const [visible, setVisible] = useState<boolean>(false);
+  const [error, setError] = useState<string | null>(null);
+
+  const fetchAvatar = () => {
```
Comment (VitalikMelnyk) [inline homeworks/stanislaw-krolikiewicz/18-react-ui/src/components/Tile/Component.tsx:17]: Also, rewrite with async/await approach.

**Review by VitalikMelnyk — CHANGES_REQUESTED** [review]
Hello @stkrolikiewicz!
I have checked your PR and left some comments.

**Comment by stkrolikiewicz** [comment]
@VitalikMelnyk I've provided suggested changes. I also changed the structure a little bit. Everything is described inside commit message.

File: homeworks/stanislaw-krolikiewicz/18-react-ui/src/views/Main/Component.tsx [hunk 6]
```diff
@@ -0,0 +1,75 @@
+import {Tile, Loader, Error as MyError} from '@/components';
+import {TilesGrid} from './_components';
+import {ReactElement, useEffect, useState} from 'react';
+import {Avatar} from '@/types';
+import {AnimatePresence} from 'framer-motion';
+import {API_URL_SINGLE_AVATAR, API_URL_MULTIPLE_AVATARS} from '@/constants';
+
+export default function Main() {
```
Comment (VitalikMelnyk) [inline homeworks/stanislaw-krolikiewicz/18-react-ui/src/views/Main/Component.tsx:8]: Please use one approach with promises. 
I recommend using everything in one async/await approach and not combining async/await with then/catch.

**Review by VitalikMelnyk — CHANGES_REQUESTED** [review]

**Review by VitalikMelnyk — APPROVED** [review]
Approved.

**Pull request merged**